manim/typing.py

```diff
@@ -1,133 +1,462 @@
+Red light, Green light, Blue light, and Alpha (opacity).
+"""
 
-# Point Types
+
+"""
+[CATEGORY]
+Point types
+"""
 
 PointDType: TypeAlias = ManimFloat
-""" DType for all points. """
+"""Default type for arrays representing points: a double-precision floating point value."""
 
 InternalPoint2D: TypeAlias = npt.NDArray[PointDType]
-""" `shape: (2,)` A 2D point. `[float, float]`.
-This type alias is mostly made available for internal use and only includes the numpy type.
+"""``shape: (2,)``
+A 2D point: ``[float, float]``.
+This type alias is mostly made available for internal use, and only includes the NumPy type.
+"""
+
+Point2D: TypeAlias = Union[InternalPoint2D, tuple[float, float]]
+"""``shape: (2,)``
+A 2D point: ``[float, float]``.
+Normally a function or method which expects a `Point2D` as a parameter can handle being
+passed a `Point3D` instead.
 """
 
-Point2D: TypeAlias = Union[InternalPoint2D, Tuple[float, float]]
-""" `shape: (2,)` A 2D point. `[float, float]`. """
+InternalPoint2D_Array: TypeAlias = npt.NDArray[PointDType]
+"""``shape: (N, 3)``
+An array of `Point2D` objects: ``[[float, float], ...]``.
+This type alias is mostly made available for internal use, and only includes the NumPy type.
+"""
+
+Point2D_Array: TypeAlias = Union[InternalPoint2D_Array, tuple[Point2D, ...]]
+"""``shape: (N, 2)``
+An array of `Point2D` objects: ``[[float, float], ...]``.
+Normally a function or method which expects a `Point2D_Array` as a parameter can handle being
+passed a `Point3D_Array` instead.
+(Please refer to the documentation of the function you are using for further type information.)
+"""
 
 InternalPoint3D: TypeAlias = npt.NDArray[PointDType]
-""" `shape: (3,)` A 3D point. `[float, float, float]`.
-This type alias is mostly made available for internal use and only includes the numpy type.
+"""``shape: (3,)``
+A 3D point: ``[float, float, float]``.
+This type alias is mostly made available for internal use, and only includes the NumPy type.
 """
 
-Point3D: TypeAlias = Union[InternalPoint3D, Tuple[float, float, float]]
-""" `shape: (3,)` A 3D point. `[float, float, float]` """
+Point3D: TypeAlias = Union[InternalPoint3D, tuple[float, float, float]]
+"""``shape: (3,)``
+A 3D point: ``[float, float, float]``.
+"""
+
+InternalPoint3D_Array: TypeAlias = npt.NDArray[PointDType]
+"""``shape: (N, 3)``
+An array of `Point3D` objects: ``[[float, float, float], ...]``.
+This type alias is mostly made available for internal use, and only includes the NumPy type.
+"""
 
-# Bezier Types
-QuadraticBezierPoints: TypeAlias = npt.NDArray[PointDType]
-""" `shape: (3,3)` An Array of Quadratic Bezier Handles `[[float, float, float], [float, float, float], [float, float, float]]`. """
+Point3D_Array: TypeAlias = Union[InternalPoint3D_Array, tuple[Point3D, ...]]
+"""``shape: (N, 3)``
+An array of `Point3D` objects: ``[[float, float, float], ...]``.
+(Please refer to the documentation of the function you are using for further type information.)
```

Comment:
Like with the `Zeros` type alias, these comments were already there when we started this PR.
I don't really know whether this comment is necessary, but because I'd prefer to keep it just in case, I removed the parentheses :+1: